pom.xml

```diff
@@ -268,6 +273,36 @@
                 <artifactId>hamcrest</artifactId>
                 <version>${hamcrest.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.hamcrest</groupId>
+                <artifactId>hamcrest-core</artifactId>
+                <version>${hamcrest.version}</version>
+            </dependency>
```

Comment:
I'm not seeing any usages of `org.hamcrest` APIs in this PR do we really need it here?

Reply:
> I'm not seeing any usages of `org.hamcrest` APIs in this PR do we really need it here?

I keep getting a used undeclared dependency in the `integration-tests` module, although I couldn't find any direct usage.

Reply:
Oh ok. Let's leave it in and I'll go digging once it's merged